neqo-transport/src/cc/classic_cc.rs

```diff
@@ -978,131 +986,184 @@ mod tests {
 
         let mut cc = ClassicCongestionControl::new(NewReno::default());
+        let mut now = now();
 
         // Change state to congestion avoidance by introducing loss.
 
         let p_lost = SentPacket::new(
             PacketType::Short,
             1,                 // pn
-            now(),             // time sent
+            now,               // time sent
             true,              // ack eliciting
             Vec::new(),        // tokens
             MAX_DATAGRAM_SIZE, // size
         );
         cc.on_packet_sent(&p_lost);
         cwnd_is_default(&cc);
-        cc.on_packets_lost(Some(now()), None, PTO, &[p_lost]);
+        now += PTO;
+        cc.on_packets_lost(Some(now), None, PTO, &[p_lost]);
         cwnd_is_halved(&cc);
         let p_not_lost = SentPacket::new(
             PacketType::Short,
-            1,                 // pn
-            now(),             // time sent
+            2,                 // pn
+            now,               // time sent
             true,              // ack eliciting
             Vec::new(),        // tokens
             MAX_DATAGRAM_SIZE, // size
         );
         cc.on_packet_sent(&p_not_lost);
-        cc.on_packets_acked(&[p_not_lost], RTT, now());
+        now += RTT;
+        cc.on_packets_acked(&[p_not_lost], RTT, now);
         cwnd_is_halved(&cc);
         // cc is app limited therefore cwnd in not increased.
         assert_eq!(cc.acked_bytes, 0);
 
         // Now we are in the congestion avoidance state.
+        assert_eq!(cc.state, State::CongestionAvoidance);
+        // simulate packet bursts below app_limit
+        let mut next_pn = 3;
+        for packet_burst_size in 1..=BELOW_APP_LIMIT_PKTS {
+            // always stay below app_limit during sent.
+            let mut pkts = Vec::new();
+            for _ in 0..packet_burst_size {
+                let p = SentPacket::new(
+                    PacketType::Short,
+                    next_pn,           // pn
+                    now,               // time sent
+                    true,              // ack eliciting
+                    Vec::new(),        // tokens
+                    MAX_DATAGRAM_SIZE, // size
+                );
+                next_pn += 1;
+                cc.on_packet_sent(&p);
+                pkts.push(p);
+            }
+            assert_eq!(cc.bytes_in_flight(), packet_burst_size * MAX_DATAGRAM_SIZE);
+            now += RTT;
+            for (i, pkt) in pkts.into_iter().enumerate() {
+                cc.on_packets_acked(&[pkt], RTT, now);
+
+                assert_eq!(
+                    cc.bytes_in_flight(),
+                    (packet_burst_size - i - 1) * MAX_DATAGRAM_SIZE
+                );
+                cwnd_is_halved(&cc); // CWND doesn't grow because we're app limited
+                assert_eq!(cc.acked_bytes, 0);
+            }
+        }
+
+        // Utilize congestion window and increase it for each ACKs arriving
```

Comment:
I don't know what this comment means, particularly "Utilize".  Do you intend to say that you are now looking to increase the congestion window by having packets sent and then acknowledged?

Reply:
Utilize is meant as opposite of underutilized from the rfc: https://datatracker.ietf.org/doc/html/rfc9002#name-underutilizing-the-congesti